hydra/_internal/config_loader_impl.py

```diff
@@ -522,6 +522,21 @@ def _compose_config_from_defaults_list(
                         f"In '{default.config_path}': Validation error while composing config:\n{e}"
                     ).with_traceback(sys.exc_info()[2])
 
+        # # remove remaining defaults lists from all nodes.
+        def strip_defaults(cfg: Any) -> None:
+            if isinstance(cfg, DictConfig):
+                if cfg._is_missing() or cfg._is_none():
+                    return
+                with flag_override(cfg, ["readonly", "struct"], False):
+                    if getattr(cfg, "__REMOVE_TOP_LEVEL_DEFAULTS__", False):
+                        cfg.pop("defaults", None)
```

Comment:
is it possible to have defaults not in cfg yet `__REMOVE_TOP_LEVEL_DEFAULTS__ = True`, that would sound like a bug?

Reply:
If a user decides to add __REMOVE_TOP_LEVEL_DEFAULTS__ to their config.